src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,242 @@
+  const options: simpleGit.SimpleGitOptions = {
+    baseDir: process.cwd(),
+    binary: 'git',
+    // We are invoking at most 3 git commands at the same time.
+    maxConcurrentProcesses: 3,
+  }
+
+  return simpleGit.gitP(options)
+}
+
+// Returns the remote of the current repository.
+export const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
+  const remotes = await git.getRemotes(true)
+  if (remotes.length === 0) {
+    throw new Error('No git remotes available')
+  }
+
+  for (const remote of remotes) {
+    // We're trying to pick the remote called with the default git name 'origin'.
+    if (remote.name === 'origin') {
+      return remote.refs.push
+    }
+  }
+
+  // Falling back to picking the first remote in the list if 'origin' is not found.
+  return remotes[0].refs.push
+}
+
+// StripCredentials removes credentials from a remote HTTP url.
+export const stripCredentials = (remote: string) => {
+  try {
+    const url = new URL(remote)
+    url.username = ''
+    url.password = ''
+
+    return url.toString()
+  } catch {
+    return remote
+  }
+}
+
+// Returns the hash of the current repository.
+const gitHash = async (git: simpleGit.SimpleGit): Promise<string> => git.revparse('HEAD')
+
+// Returns the tracked files of the current repository.
+export const gitTrackedFiles = async (git: simpleGit.SimpleGit): Promise<string[]> => {
+  const files = await git.raw('ls-files')
+
+  return files.split(/\r\n|\r|\n/)
+}
+
+// Trims from a set of characters from the start of a string.
+const trimStart = (str: string, chars: string[]) => {
+  let start = 0
+  const end = str.length
+  while (start < end && chars.indexOf(str[start]) >= 0) {
+    ++start
+  }
+
+  return start > 0 ? str.substring(start, end) : str
+}
+
+// Trims from a set of characters from a string.
+const trim = (str: string, chars: string[]) => {
+  let start = 0
+  let end = str.length
+  while (start < end && chars.indexOf(str[start]) >= 0) {
+    ++start
+  }
+  while (end > start && chars.indexOf(str[end - 1]) >= 0) {
+    --end
+  }
+
+  return start > 0 || end < str.length ? str.substring(start, end) : str
+}
+
+// Generates a proper source file path from a sourcemap:
+// - Strip a set of hard-coded prefixes ('webpack:///./')
+// - Strip the eventual projectPath
+// - Removes query parameters
+//
+// It returns the new source as well as whether or not the specified projectPath was stripped.
+//
+// For example, the following source path:
+// webpack:///./project/folder1/folder2/src.js?abc123
+//
+// Will be cleaned up into 'project/folder1/folder2/src.js'.
+export const cleanupSource = (source: string, projectPath: string): [string, boolean] => {
+  // Prefixes
+  const prefixesToRemove = ['webpack:']
+  for (const p of prefixesToRemove) {
+    if (source.startsWith(p)) {
+      source = source.slice(p.length)
+    }
+  }
+  source = trimStart(source, ['/', '.'])
+  // ProjectPath
+  projectPath = trim(projectPath, ['/', '.'])
```

Comment:
This will remove `/../` and this will transform `/.hidden` into `hidden`. Is that what we want?

Reply:
Probably not. I updated the code and added some tests to cover those edge cases.

Reply:
It is now covered in e832d79 cc @lraucy

Reply:
This should not be relevant anymore with the changes in e832d79 cc @lraucy